tools/clang/lib/Parse/ParseDecl.cpp

```diff
@@ -2202,6 +2202,17 @@ Parser::DeclGroupPtrTy Parser::ParseDeclGroup(ParsingDeclSpec &DS,
     return DeclGroupPtrTy();
   }
 
+  // HLSL changes begin
+  // Initialize the default matrix orientation.
+  if (!Parser::Actions.PackMatrixRowMajorPragmaOn && !Parser::Actions.PackMatrixColMajorPragmaOn) {
+    DS.SetDefaultMatrixOrientation(true);
```

Comment:
It's a little confusing that this doesn't take the global setting into account.  I know that for now, we only read the flag if the pragmas are set, but the method and flag don't make that clear.  Should we add a TODO comment here at least that we intend to incorporate the global flag in a subsequent change?

Additionally, if/when we do treat them both the same way, I think it would help to have an accessor for retrieving the current default matrix orientation, hiding the details of the command line option and the pragmas.

Reply:
Addded TODO.